internal/pkg/tablemgr/routing_table_manager.go

```diff
@@ -232,10 +232,12 @@ func (r *DefaultRoutingTableManager) RouteEvent(ctx context.Context, tid tenant.
 		event.WithTracePayloadOnNack(false),
 	)
 	if err != nil {
-		return errors.New("bad test event for route " + routeId)
+		return "", errors.New("bad test event for route " + routeId)
 	}
+	traceId, _, _ := e.GetPathValue("trace.id")
```

Comment:
Have u try?
```
traceId, ok := e.Context().Value(rtsemconv.EarsLogTraceIdKey).(string)
```

Reply:
Also, i wonder if we provide trace Id in the API call header? Does it propagate?